src/Features/Blockcore.Features.Wallet/Types/WalletStore.cs

```diff
@@ -0,0 +1,205 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using Blockcore.Configuration;
+using Blockcore.Features.Wallet.Exceptions;
+using Blockcore.Utilities;
+using Blockcore.Utilities.JsonConverters;
+using LiteDB;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using NBitcoin.DataEncoders;
+using Newtonsoft.Json;
+
+namespace Blockcore.Features.Wallet.Types
+{
+    public interface IWalletStore
+    {
+        void InsertOrUpdate(TransactionData item);
+
+        IEnumerable<TransactionData> GetForAddress(string address);
+
+        int CountForAddress(string address);
+
+        TransactionData GetForOutput(OutPoint outPoint);
+
+        bool Remove(OutPoint outPoint);
+
+        WalletData GetData();
+
+        void SetData(WalletData data);
+    }
+
+    public class WalletData
+    {
+        [BsonId]
+        public string Key { get; set; }
+
+        public string EncryptedSeed { get; set; }
+
+        public string WalletName { get; set; }
+
+        public HashHeightPair WalletTip { get; set; }
+
+        public ICollection<uint256> BlockLocator { get; set; }
+    }
+
+    public class WalletStore : IWalletStore, IDisposable
+    {
+        private LiteDatabase db;
+        private readonly Network network;
+        private LiteCollection<WalletData> dataCol;
+        private LiteCollection<TransactionData> trxCol;
+
+        public WalletData WalletData { get; private set; }
+
+        public BsonMapper Mapper => this.db.Mapper;
+
+        public WalletStore(Network network, DataFolder dataFolder, Wallet wallet)
+        {
+            string dbPath = Path.Combine(dataFolder.WalletFolderPath, $"{wallet.Name}.txdb.litedb");
+
+            if (!Directory.Exists(dataFolder.WalletFolderPath))
+            {
+                Directory.CreateDirectory(dataFolder.WalletFolderPath);
+            }
+
+            BsonMapper mapper = this.Create();
+            LiteDB.FileMode fileMode = RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? LiteDB.FileMode.Exclusive : LiteDB.FileMode.Shared;
```

Comment:
Should have a utility method to check OS, something for the future.

Reply:
And could be a method in this store object that returns "RequiredLockMode()"

Reply:
I actually used exclusive always because it has better performance (shared opens and closes connection on each call)

Reply:
I just made all wallets mode direct, better perf.